controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/DeploymentTrigger.java

```diff
@@ -376,13 +383,18 @@ private boolean changeDeployed(Application application, JobType job) {
         if (deployment == null)
             return false;
 
-        if (application.change().platform().filter(version -> version.isAfter(deployment.version())).isPresent())
-            return false;
+        int applicationComparison = application.change().application()
+                                               .map(version -> version.compareTo(deployment.applicationVersion()))
+                                               .orElse(0);
 
-        if (application.change().application().filter(version -> version.compareTo(deployment.applicationVersion()) > 0).isPresent())
-            return false;
+        int platformComparion = application.change().platform()
+                                           .map(version -> version.compareTo(deployment.version()))
+                                           .orElse(0);
 
-        return true;
+        if (applicationComparison == -1 || platformComparion == -1)
+            return true; // Avoid downgrades!
+
+        return applicationComparison == 0 && platformComparion == 0;
```

Comment:
Previous implementation would allow a downgrade of application if platform was an upgrade >_<